src/Renci.SshNet/Common/SshDataStream.cs

```diff
@@ -56,20 +56,47 @@ public bool IsEndOfData
             }
         }
 
+#if NET462 || NETSTANDARD2_0
+        private int Read(Span<byte> buffer)
+        {
+            var sharedBuffer = System.Buffers.ArrayPool<byte>.Shared.Rent(buffer.Length);
+            try
+            {
+                var numRead = Read(sharedBuffer, 0, buffer.Length);
+
+                new ReadOnlySpan<byte>(sharedBuffer, 0, numRead).CopyTo(buffer);
+                return numRead;
+            }
+            finally
+            {
+                System.Buffers.ArrayPool<byte>.Shared.Return(sharedBuffer);
+            }
```

Comment:
Let's remove the try-finally. It is generally considered not worthwhile - see here for some discussion https://github.com/dotnet/runtime/issues/71249#issuecomment-1165597563

Obviously it doesn't really matter here - just establishing a convention